client/flutter/lib/main.dart

```diff
@@ -73,6 +74,24 @@ class _MyAppState extends State<MyApp> {
       },
     );
 
+    updateFCMToken();
+  }
+
+  // Ask firebase for a token on every launch.
+  // If the token is different from what we have stored...update it.
+  updateFCMToken() async {
+    // Only check if notifications are enabled. 
+    bool notificationsEnabled =
+        await UserPreferences().getNotificationsEnabled();
+
+    if (notificationsEnabled) {
+      final token = await _firebaseMessaging.getToken();
+      String storedToken = await UserPreferences().getFCMToken();
```

Comment:
Could we use `final` here too?

```suggestion
      final storedToken = await UserPreferences().getFCMToken();
```